kvmagent/kvmagent/plugins/vm_plugin.py

```
@@ -10819,6 +10834,111 @@ def find_mdev_device(vm_uuid, mdev_uuid):
+            raise kvmagent.KvmError('invalid dGPU shmem path[%s], path must be normalized under /dev/shm' % mem_path)
+        shmem_size_mb = _normalize_dgpu_shmem_size_mb(shmem_size)
+
+        shmem_name = os.path.basename(normalized_mem_path)
+        root = etree.Element('shmem', {'name': shmem_name})
+        e(root, 'model', attrib={'type': 'ivshmem-plain'})
+        e(root, 'size', str(shmem_size_mb), attrib={'unit': 'M'})
+        return etree.tostring(root, encoding="unicode"), normalized_mem_path, shmem_name
+
+    def _is_dgpu_shmem_attached(self, vm, shmem_name):
+        domain_xml = vm.domain.XMLDesc(0)
+        root = etree.fromstring(domain_xml)
```

Ruff (0.15.12)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)